### PR TITLE
[otbn] document/comment modexp otbn assembly lib

### PR DESCRIPTION
**Pull request by felixmiller**
Document modexp otbn assembly code

This finishes #2908

### CODE REVIEW

File: sw/otbn/code-snippets/modexp.S [hunk 1]
```diff
@@ -203,80 +626,324 @@ setupPtrs:
...
+  /* prepare pointers and other parameters */
   jal       x1, setupPtrs
+
+  /* call montmul(1,A) algorithm */
   jal       x1, mul1_exp
+
   ecall
 
+
+/* Constant-time Mongomery Modular Multiplication
```
Comment (rswarbrick) [inline sw/otbn/code-snippets/modexp.S:910]: Nit: "Montgomery"

File: sw/otbn/code-snippets/modexp.S [hunk 2]
```diff
@@ -55,38 +193,105 @@ computeRR:
...
+    /* reset pointer to 1st limb of bigint in regfile */
+    li        x8, 5
+
+    /* reset pointer to modulus in dmem */
+    lw        x16, 0(x0)
+
+    /* reset flags of FG1 */
+    bn.sub    w3, w3, w3, FG1
+
+    /* compare intermediate biginit y with modulus
```
Comment (rswarbrick) [inline sw/otbn/code-snippets/modexp.S:260]: Nit: bigint

File: sw/otbn/code-snippets/modexp.S [hunk 3]
```diff
@@ -319,41 +1046,164 @@ mulx_exp:
...
+    /* conditional select: w2 = FG0.C?w[x8+i]:dmem[x21+i] */
+    bn.sel    w2, w0, w3, C
+
+    /* store selected limb to dmem */
+    bn.sid    x11, 0(x21++)
+
   ret
 
+
+/* Constant-time biginit modular exponentiation
```
Comment (rswarbrick) [inline sw/otbn/code-snippets/modexp.S:1133]: Nit: biginit

**Review by rswarbrick — APPROVED** [review]
Wow, this is detailed! I really like all the comments, and feel like I understand things a bit better now, which is nice.

As well as the nitty typos I noticed, I think you've written "Clobbered Flag Groups" with capitals in the second half of the file and without in the first. (Yes, it's really pedantic, but I thought you'd like to know that someone actually read this :-D)

**Review by imphil — APPROVED** [review]
Great work Felix, thank you! 

On comments: You sometimes go for doxygen-style comments (`/**`), and sometimes for normal block comments (`/*`). Both is fine from my PoV, but it would be nice to make it consistent.

**Comment by felixmiller** [comment]
Thanks @rswarbrick and @imphil! Pushed an update addressing your comments. If there is more please don't hesitate to point it out. This is a text-heavy commit, so if there is anything w.r.t. to terminology, style, use of English language, letter case, etc., happy to improve/correct.

**Pull request merged**